shell/spy.go

```diff
@@ -1,19 +1,32 @@
 package shell
 
 import (
-	"bytes"
+	"fmt"
+	"io"
 	"log"
 	"os/exec"
+	"strings"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
-	var out bytes.Buffer
-	cmd.Stdout = &out
+// Parser represents a function that parses string text
+// and returns a tring and slice of strings
+type Parser func(string) (string, []string)
```

Comment:
Great, but it seems like the only reason we need this type is so that we can pass a `Parser` to `Execute`. But why would we need to do that? Only so we could inject a fake from a test. But we don't do that! So `Execute` could just call `ParseCommand` directly, couldn't it?